### PR TITLE
Clean up "make all" for 5.2 branch

### LINKED ISSUE #2418: Get make all clean on the ctsm5.2 branch
### Brief summary of bug

``` bash
cd python
make all
```
is not clean on the ctsm5.2.mksurfdata branch. We need this working to bring to master.

### General bug information

**CTSM version you are using:** alpha-ctsm5.2.mksrf.23_ctsm5.1.dev171-5-g76217f249 (although it predates this)

**Does this bug cause significantly incorrect results in the model's science?** No
**Configurations affected:** Just python tools testing

### Details of bug

The python sys tests fail as well as the python lint checking.

### Important output or errors that show the problem
make stest fails..
``` text
...E
Stdout:
---- building a base case -------
---- creating a base case -------
---- base case created ------
---- base case setup ------
using this version: latest
---- cloning the base case in /glade/derecho/scratch/erik/tmp/tmp28jgfdnd/BART.transient

======================================================================
ERROR: test_one_site (test.test_sys_run_neon.TestSysRunNeon)
This test specifies a site to run
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/glade/work/erik/ctsm_worktrees/ctsm5.2.mksurfdata/python/ctsm/test/test_sys_run_neon.py", line 57, in test_one_site
    main("")
  File "/glade/work/erik/ctsm_worktrees/ctsm5.2.mksurfdata/python/ctsm/site_and_regional/run_neon.py", line 238, in main
    experiment,
  File "/glade/work/erik/ctsm_worktrees/ctsm5.2.mksurfdata/python/ctsm/site_and_regional/neon_site.py", line 312, in run_case
    case.create_namelists()
  File "/glade/work/erik/ctsm_worktrees/ctsm5.2.mksurfdata/cime/CIME/case/preview_namelists.py", line 97, in create_namelists
    case=self,
  File "/glade/work/erik/ctsm_worktrees/ctsm5.2.mksurfdata/cime/CIME/utils.py", line 665, in import_and_run_sub_or_cmd
    raise e1 from None
  File "/glade/work/erik/ctsm_worktrees/ctsm5.2.mksurfdata/cime/CIME/utils.py", line 662, in import_and_run_sub_or_cmd
    cmd, cmdargs, subname, subargs, logfile, case, from_dir, timeout
  File "/glade/work/erik/ctsm_worktrees/ctsm5.2.mksurfdata/cime/CIME/utils.py", line 706, in run_sub_or_cmd
    getattr(mod, subname)(*subargs)
  File "/glade/work/erik/ctsm_worktrees/ctsm5.2.mksurfdata/cime_config/buildnml", line 372, in buildnml
    expect(rc == 0, "Command %s failed rc=%d\nout=%s\nerr=%s" % (cmd, rc, out, err))
  File "/glade/work/erik/ctsm_worktrees/ctsm5.2.mksurfdata/cime/CIME/utils.py", line 175, in expect
    raise exc_type(msg)
CIME.utils.CIMEError: ERROR: Command /glade/work/erik/ctsm_worktrees/ctsm5.2.mksurfdata/bld/build-namelist failed rc=255
out=
err=ERROR : CLM build-namelist::CLMBuildNamelist::setup_logic_lnd_tuning() : lnd_tuning_mode has a value (clm6_0_GSWP3v1) that is NOT valid. Valid values are: 'clm4_5_CRUv7' 'clm4_5_GSWP3v1' 'clm4_5_cam6.0' 'clm5_0_cam6.0' 'clm5_0_CRUv7' 'clm5_0_GSWP3v1' 'clm5_1_GSWP3v1' 'clm5_1_cam6.0'

Stdout:
---- building a base case -------
---- creating a base case -------
---- base case created ------
---- base case setup ------
using this version: latest
---- cloning the base case in /glade/derecho/scratch/erik/tmp/tmp28jgfdnd/BART.transient

----------------------------------------------------------------------
Ran 33 tests in 52.550s

FAILED (errors=1)
(ctsm_pylib) ctsm
```
make lint fails...
``` text
pylint -j 4 --rcfile=ctsm/.pylintrc --fail-under=0 ctsm
************* Module ctsm.test.test_sys_gen_mksurfdata_jobscript_multi
ctsm/test/test_sys_gen_mksurfdata_jobscript_multi.py:50:4: W0102: Dangerous default value [] as argument (dangerous-default-value)
************* Module ctsm.test.test_sys_gen_mksurfdata_jobscript_single
ctsm/test/test_sys_gen_mksurfdata_jobscript_single.py:53:4: W0102: Dangerous default value [] as argument (dangerous-default-value)
************* Module ctsm.test.test_sys_mesh_modifier
ctsm/test/test_sys_mesh_modifier.py:47:8: W0612: Unused variable 'fsurdat_in' (unused-variable)
ctsm/test/test_sys_mesh_modifier.py:150:8: W0201: Attribute 'fsurdat_in' defined outside __init__ (attribute-defined-outside-init)
ctsm/test/test_sys_mesh_modifier.py:188:8: W0201: Attribute 'fsurdat_in' defined outside __init__ (attribute-defined-outside-init)
************* Module ctsm.test.test_unit_modify_fsurdat
ctsm/test/test_unit_modify_fsurdat.py:393:0: C0301: Line too long (106/100) (line-too-long)
************* Module ctsm.toolchain.gen_mksurfdata_jobscript_single
ctsm/toolchain/gen_mksurfdata_jobscript_single.py:133:0: C0301: Line too long (103/100) (line-too-long)
ctsm/toolchain/gen_mksurfdata_jobscript_single.py:96:0: R0915: Too many statements (68/50) (too-many-statements)
ctsm/toolchain/gen_mksurfdata_jobscript_single.py:7:0: W0611: Unused import sys (unused-import)
ctsm/toolchain/gen_mksurfdata_jobscript_single.py:15:0: C0411: third party import "from CIME.XML.env_mach_specific import EnvMachSpecific" should be placed before "from ctsm import add_cime_to_path" (wrong-import-order)
ctsm/toolchain/gen_mksurfdata_jobscript_single.py:16:0: C0411: third party import "from CIME.BuildTools.configure import FakeCase" should be placed before "from ctsm import add_cime_to_path" (wrong-import-order)
************* Module ctsm.toolchain.gen_mksurfdata_jobscript_multi
ctsm/toolchain/gen_mksurfdata_jobscript_multi.py:171:0: C0301: Line too long (105/100) (line-too-long)
ctsm/toolchain/gen_mksurfdata_jobscript_multi.py:249:0: C0301: Line too long (148/100) (line-too-long)
ctsm/toolchain/gen_mksurfdata_jobscript_multi.py:373:0: C0301: Line too long (127/100) (line-too-long)
ctsm/toolchain/gen_mksurfdata_jobscript_multi.py:399:0: C0301: Line too long (109/100) (line-too-long)
ctsm/toolchain/gen_mksurfdata_jobscript_multi.py:402:0: C0301: Line too long (101/100) (line-too-long)
ctsm/toolchain/gen_mksurfdata_jobscript_multi.py:405:0: C0301: Line too long (103/100) (line-too-long)
ctsm/toolchain/gen_mksurfdata_jobscript_multi.py:409:8: W0612: Unused variable 'n_p' (unused-variable)
ctsm/toolchain/gen_mksurfdata_jobscript_multi.py:140:0: R0915: Too many statements (61/50) (too-many-statements)
************* Module ctsm.toolchain.gen_mksurfdata_namelist
ctsm/toolchain/gen_mksurfdata_namelist.py:300:20: E1101: Module 'netCDF4' has no 'Dataset' member (no-member)
ctsm/toolchain/gen_mksurfdata_namelist.py:258:0: R0915: Too many statements (335/50) (too-many-statements)

------------------------------------------------------------------
Your code has been rated at 9.97/10 (previous run: 9.97/10, +0.00)
```

### PR DESCRIPTION
**Pull request by samsrabin**
### Description of changes

Resolves `pylint` complaints for 5.2 branch (PR #2372) described in Issue #2418. Note, @ekluzek, I was unable to reproduce the system test failure you described in that issue.

### Specific notes

**Contributors other than yourself, if any:** None

**CTSM Issues Fixed:**
* Resolves #2418

**Are answers expected to change (and if so in what way)?** No

**Any User Interface Changes (namelist or namelist defaults changes)?** No

**Testing performed, if any:** `make all` is now clean.

### CODE REVIEW

**Comment by samsrabin** [comment]
@ekluzek The system test failure you described in #2418 seems to be related to your addition of `clm6_0` physics in #2417. I think it'd be better if you made a task in that issue to resolve it. That way, this PR can close #2418.

**Review by slevis-lmwg — COMMENTED** [review]
Wow, this looks like a nice cleanup @samsrabin, thank you!

Given that the changes feel extensive (at least to me), and that we are about to finish ctsm5.2, it's important to confirm that "make all" doesn't now break elsewhere and that clm_pymods still works. And then maybe run an example of 
```
./gen_mksurfdata_namelist
./gen_mksurfdata_jobscript_single
```
and one of 
```
./gen_mksurfdata_jobscript_multi
```
(no need to generate fsurdat files) to confirm that answers have not changed. I'm happy to meet if that helps.

**Comment by ekluzek** [comment]
@slevis-lmwg and @samsrabin. @slevis-lmwg I think you are right that we need to do more testing here. But, I'm not sure that @samsrabin should do all of it. He did volunteer his time for this and I appreciate his efforts. So I think @slevis-lmwg and I should do more testing. But, I'll leave it up to @samsrabin if you are up to do more testing for us...

**Comment by ekluzek** [comment]
Also note @slevis-lmwg there are python tests for the three main scripts that you give above. So we only need to maybe do some testing on something that wouldn't be caught with those tests.

I ran simple tests for above from the main tool in tools/mksurfdata_esmf:

./gen_mksurfdata_namelist --start-year 1850 --end-year 2000 --res 0.9x1.25
./gen_mksurfdata_build.sh
./gen_mksurfdata_jobscript_single --number-of-nodes 10 --tasks-per-node 128 --namelist-file surfdata_0.9x1.25_hist_1850_78pfts_c240314.namelist./gen_mksurfdata_jobscript_multi --number-of-nodes 10 --scenario global-present

And compared the results and they are the same other than the paths to where the CTSM clone is.

**Comment by ekluzek** [comment]
I reviewed the testing being done compared to what we actually create using the Makefile. The jobscript_single and jobscript_multi are sufficiently simple that there doesn't look to be anything else that needs to be tested for them. The namelist script is the one with the complexity. The things that need to be done there are: simple, potveg, vic, hist, and SSP. And those are being tested in the python tests, so I'd say we are all good here.

This is an example of having sufficient testing that we can refactor the code without having fear that it's being screwed up. That's one of the benefits of having good testing in place.

I did run clm_pymods for the baseline and this branch, and they compare the same. There is one test that initially failed in both, that we should look into, but it acted the same as the baseline.

MKSURFDATAESMF_P64x1.f10_f10_mg37.I1850Clm50BgcCrop.derecho_intel

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (ekluzek) [inline python/ctsm/toolchain/gen_mksurfdata_namelist.py:439]: Nice natural subroutinization here.

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (ekluzek) [inline python/ctsm/toolchain/gen_mksurfdata_namelist.py:462]: Again a nice natural break here.

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (ekluzek) [inline python/ctsm/toolchain/gen_mksurfdata_namelist.py:482]: Another natural subroutine nice.

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (ekluzek) [inline python/ctsm/toolchain/gen_mksurfdata_namelist.py:533]: Natural subroutine here.

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (ekluzek) [inline python/ctsm/toolchain/gen_mksurfdata_namelist.py:572]: Natural subroutine.

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (ekluzek) [inline python/ctsm/toolchain/gen_mksurfdata_namelist.py:613]: Another subroutine. This must be the raw input data files.

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (ekluzek) [inline python/ctsm/toolchain/gen_mksurfdata_namelist.py:703]: You got rid of the leading underscore here. And I think that makes sense here. That's only when it's meant to be private and since this is now what's returned for the new function -- this makes sense.

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (ekluzek) [inline python/ctsm/toolchain/gen_mksurfdata_namelist.py:703]: Another natural function.

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (ekluzek) [inline python/ctsm/toolchain/gen_mksurfdata_namelist.py:804]: Another part that could be made shorter in the future.

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (ekluzek) [inline python/ctsm/toolchain/gen_mksurfdata_namelist.py:628]: This one must be really close to the limit. Perhaps we should increase it?

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (ekluzek) [inline python/ctsm/toolchain/gen_mksurfdata_namelist.py:892]: I'm not sure what it's complaining about here. I can look it up, but if you can easily explain that would be great.

**Review by ekluzek — APPROVED** [review]
Nice work. And quick work. I'm thinking you should show us your process here.

This is great. I have a few questions, but nothing important to change.

I checked the number of pylint disable statements and they are reasonably few. I do point out that there are only two "too many statements" now, which is a great reduction. Eventually we should shorten that, but that's such an improvement already.

File: python/ctsm/test/test_sys_mesh_modifier.py
Comment (samsrabin) [reply python/ctsm/test/test_sys_mesh_modifier.py:71]: `pylint` didn't like something in `self` being defined outside the initialization function. That was the genesis of these complaints:
```
ctsm/test/test_sys_mesh_modifier.py:150:8: W0201: Attribute 'fsurdat_in' defined outside __init__ (attribute-defined-outside-init)
ctsm/test/test_sys_mesh_modifier.py:188:8: W0201: Attribute 'fsurdat_in' defined outside __init__ (attribute-defined-outside-init)
```

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (samsrabin) [reply python/ctsm/toolchain/gen_mksurfdata_namelist.py:892]: I can't explain it, honestly. It thinks the `netCDF4` module doesn't have a member called `Dataset`, but when I open a Python console in `ctsm_pylib` and do `from netCDF4 import Dataset` it works fine.

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (samsrabin) [reply python/ctsm/toolchain/gen_mksurfdata_namelist.py:628]: I think I like the limit being kept at 50, and then we can have a little bit of wiggle room. If we increase it, we'll always be tempting to add wiggle room on top of that.

Comment (ekluzek) [reply python/ctsm/toolchain/gen_mksurfdata_namelist.py:628]: Yes, the sliding down the hill problem, it keeps sliding toward infinity. I was thinking the same. I suppose 50 is typically a lot for python. It's a guideline for having code that's too complex and can't be unit tested. So yes the longer we make it the longer we'll want to keep making it. I suppose that default of 50 has something behind it as well.

Comment (ekluzek) [reply python/ctsm/toolchain/gen_mksurfdata_namelist.py:628]: OK, resolving.

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (ekluzek) [reply python/ctsm/toolchain/gen_mksurfdata_namelist.py:892]: OK, just pylint strangeness. It might be good to mark with a comment that says above. We likely will need to use this in other places...

File: python/ctsm/test/test_sys_mesh_modifier.py
Comment (samsrabin) [reply python/ctsm/test/test_sys_mesh_modifier.py]: Fixed with aa3ddba7696036f2ac4b1808666a402db61d0f24.

**Comment by samsrabin** [comment]
Thanks for the reviews, y'all! I don't think I have time for additional testing, though. I'll let you handle that if that's alright. Feel free to go ahead and merge this when you think it's ready.

File: python/ctsm/test/test_sys_mesh_modifier.py
Comment (ekluzek) [reply python/ctsm/test/test_sys_mesh_modifier.py:71]: Ahhh OK. A solution could be to set it inside of __init__ to None, or to pass it in. I think I slightly prefer having it defined in __init__, you could then have some standard checking to make sure it's set to something besides None. Any thoughts there?

**Comment by ekluzek** [comment]
Awesome, thanks so much @samsrabin this was really great work and I'm glad to see it come in.

File: python/ctsm/test/test_sys_mesh_modifier.py
Comment (samsrabin) [reply python/ctsm/test/test_sys_mesh_modifier.py:71]: I sort of like having it outside `self` here, since it's a variable that differs depending on which test you're running. Requiring it to be specified in the test also avoids the need to check that it's not `None`, because people won't be able to accidentally not specify it.

File: python/ctsm/toolchain/gen_mksurfdata_namelist.py
Comment (samsrabin) [reply python/ctsm/toolchain/gen_mksurfdata_namelist.py:892]: Resolved with d8b6a5d80.

File: python/ctsm/test/test_sys_mesh_modifier.py
Comment (ekluzek) [reply python/ctsm/test/test_sys_mesh_modifier.py:71]: That makes sense. OK, let's leave it as it is.

**Pull request merged**
